hack/create-user-workspace.sh

```diff
@@ -22,12 +22,8 @@ then
   export KUBECONFIG=${KCP_KUBECONFIG}
 fi
 
-echo "Accessing the home workspace:"
-kubectl ws '~'
```

Comment:
We should create the appstudio/hacbs user workspace in a home workspace only. No matter what `ROOT_WORKSPACE` is set to. `ROOT_WORKSPACE` defines the place where service provider workspaces are created

Reply:
in that case we can maybe have `SP_ROOT_WORKSPACE` and `HOME_WORKSPACE`

Reply:
I would say that we should create these workspaces always in a home workspace - this is also the expected behavior/location from the HAC point of view. If we need something else in the future, then we can improve it later.
As for `SP_ROOT_WORKSPACE` - yeah, we can rename it, why not.

Reply:
Expecting that e2e-tests will need this soon, since they will be using one account

Reply:
Agree, I already detect a issue in my  local installation:

```log
application.argoproj.io/all-components patched
application.argoproj.io/dev-application-api patched
application.argoproj.io/dev-application-service patched
application.argoproj.io/dev-authorization-appstudio patched
application.argoproj.io/dev-authorization-hacbs patched
application.argoproj.io/dev-enterprise-contract patched
application.argoproj.io/dev-spi patched
application.argoproj.io/spi-vault patched
Accessing the home workspace:
Current workspace is "root:users:uz:hm:rh-sso-redhat-appstudio-qe".
Creating & accessing AppStudio workspace 'appstudio':
Workspace "appstudio" (type root:universal) already exists.
Workspace "appstudio" (type root:universal) is ready to use.
Current workspace is "root:users:uz:hm:rh-sso-redhat-appstudio-qe:appstudio".
apibinding.apis.kcp.dev/pipeline-service unchanged
Error from server (Forbidden): error when creating "STDIN": apibindings.apis.kcp.dev "application-api" is forbidden: spec.reference.workspace.path: not pointing to an ancestor or child of an ancestor of "root:users:uz:hm:rh-sso-redhat-appstudio-qe:appstudio"
Error from server (Forbidden): error when creating "STDIN": apibindings.apis.kcp.dev "application-service" is forbidden: spec.reference.workspace.path: not pointing to an ancestor or child of an ancestor of "root:users:uz:hm:rh-sso-redhat-appstudio-qe:appstudio"
Error from server (Forbidden): error when creating "STDIN": apibindings.apis.kcp.dev "spi" is forbidden: spec.reference.workspace.path: not pointing to an ancestor or child of an ancestor of "root:users:uz:hm:rh-sso-redhat-appstudio-qe:appstudio"
Accessing the home workspace:
Current workspace is "root:users:uz:hm:rh-sso-redhat-appstudio-qe".
Creating & accessing AppStudio workspace 'hacbs':
Workspace "hacbs" (type root:universal) already exists.
Workspace "hacbs" (type root:universal) is ready to use.
Current workspace is "root:users:uz:hm:rh-sso-redhat-appstudio-qe:hacbs".
apibinding.apis.kcp.dev/pipeline-service unchanged
Error from server (Forbidden): error when creating "STDIN": apibindings.apis.kcp.dev "application-api" is forbidden: spec.reference.workspace.path: not pointing to an ancestor or child of an ancestor of "root:users:uz:hm:rh-sso-redhat-appstudio-qe:hacbs"
Error from server (Forbidden): error when creating "STDIN": apibindings.apis.kcp.dev "application-service" is forbidden: spec.reference.workspace.path: not pointing to an ancestor or child of an ancestor of "root:users:uz:hm:rh-sso-redhat-appstudio-qe:hacbs"
Error from server (Forbidden): error when creating "STDIN": apibindings.apis.kcp.dev "spi" is forbidden: spec.reference.workspace.path: not pointing to an ancestor or child of an ancestor of "root:users:uz:hm:rh-sso-redhat-appstudio-qe:hacbs"
Current workspace is "root:users:uz:hm:rh-sso-redhat-appstudio-qe:ci-kvzst".

```

Reply:
Keeping default '~' but allowing to override by `HOME_WORKSPACE`, in case that it will not be supported in future we can just remove it.

Reply:
ok, fair enough :+1: